rdm/pull.py

```diff
@@ -0,0 +1,56 @@
+'''
+Functions that parse the given project management service and produce
+`requirements.yml` and `problem_reports.yml` files.  The selected project
+management backend is specified in the project configuration file.
+'''
+import os
+from getpass import getpass
+
+from github import Github
+import yaml
+
+from rdm.util import get_project_settings
+
+
+def pull_requirements_and_reports(system_yml_path):
+    settings = get_project_settings(system_yml_path)
+    if settings['software_requirements_location'] == 'GitHub':
+        output_dir = os.path.dirname(system_yml_path)
+        pull_from_github(settings, output_dir)
+    else:
+        raise ValueError("Specified software requirements location not yet supported.")
+
+
+def pull_from_github(settings, output_dir):
+    g = authenticate_github()
+    repository = g.get_repo(settings['repository'])
+
+    requirements = {issue.number: {'title': issue.title, 'description': issue.body.replace('\r', '')}
+                    for issue in repository.get_issues()
+                    if 'requirement' in issue.labels}
+    print('Found {} requirement(s)'.format(len(requirements)))
+
+    problem_reports = {issue.number: {'title': issue.title, 'description': issue.body.replace('\r', '')}
+                       for issue in repository.get_issues()
+                       if 'problem-report' in issue.labels}
+    print('Found {} problem report(s)'.format(len(problem_reports)))
+
+    requirements_path = os.path.join(output_dir, 'requirements.yml')
+    problem_reports_path = os.path.join(output_dir, 'problem_reports.yml')
+    with open(requirements_path, 'w') as f:
```

Comment:
Perhaps we should write the requirements and problem reports files in `pull_requirements_and_reports`?

I think this would keep more consistent levels of abstraction.  It seems like `pull_from_github` should handle all the Github-specific stuff, however, writing these files to disk is not Github specific.  Perhaps `pull_from_github` can return the two dicts as tuples.

Reply:
That's a good point. It would be good to maintain that level of abstraction.